File: marcenaria-java-activej/src/main/java/github/gmess/models/geometrias/Cilindro.java
package github.gmess.models.geometrias;

import github.gmess.utils.NumberUtils;

public record Cilindro(
  String geometria,
  String estrutura, 
  
  String raioBase, 

  String altura) implements Geometria {

  @Override
  public final String getGeometria() {
    return "cilindro";
  }

  @Override
  public final String getEstrutura() {
    return this.estrutura;
  }

  @Override
  public final double getArea() {
    final double raioBase = NumberUtils.centimetersToDouble(this.raioBase);
    final double altura = NumberUtils.centimetersToDouble(this.altura);

    return NumberUtils.TWO_PI * raioBase * (raioBase * altura);
  }
} 

File: MarcenariaDotNet/Dtos/Geometrias/Cilindro.cs
using System.Text.Json.Serialization;
using MarcenariaDotNet.Utils;

namespace MarcenariaDotNet.Dtos.Geometrias;

public struct Cilindro : Geometria
{

  public String Estrutura { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double RaioBase { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double Altura { get; init; }

  double Geometria.GetArea()
  {
    return NumberUtils.TWO_PI * RaioBase * (RaioBase + Altura);
  }

  string Geometria.GetEstrutura()
  {
    return Estrutura;
  }

  string Geometria.GetGeometria()
  {
     return "cilindro";
  }
}
